src/lava/magma/core/model/py/ports.py

```diff
@@ -513,6 +513,14 @@ def write(
         """
         pass
 
+    # TODO: (PP) This should be optimized by a proper CSPSendPort wait
+    def wait(self):
+        """Method which block execution until (all) previous write requests have
```

Comment:
This is a high-level user facing method. I suggest to write a proper docstring either at the RefPort or method level that explains what this method does and why the RefPort needs it.

In a final comment, I would state that this is a preliminary implementation that achieves the desired blocking behavior until the earlier WRITE has been acknowledged via a read(). This is inefficient and will be optimized later at the CspChannel level.